lib/native-client.js

```diff
@@ -723,6 +724,17 @@ class NativeClient extends EventEmitter {
   }
 
   /**
+   * Determine if we are connected to a mongos
+   *
+   * @param {Object} ismaster - The ismaster response.
+   *
+   * @returns {Boolean} If the server is a mongos.
+   */
+  _isMongos(ismaster) {
+    return ismaster.msg === SHARDED;
```

Comment:
1. Should we update the implementation of `_isWritable` just above on line 723 to say `return ismaster.ismaster === true || this._isMongos(ismaster);`?
2. Should we explicitly link to https://docs.mongodb.com/master/reference/command/isMaster/#sharded-instances in the JSDoc of `SHARDED`, or is it easy enough to find elsewhere?